### ISSUE TITLE
Barbs broken for 1D

### ISSUE THREAD

**Issue by dopplershift** [SciTools/cartopy#806]
When trying to use CartoPy's wrapper around `barbs`, if you have 1D data (like irregular points), and you try to plot in a different projection between data and map (if say you're plotting lon/lat data in a projection), then you get the following:

``` pytb
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-57-ca96cf8f7709> in <module>()
     16 ax.barbs(np.array(recon_data['lon']), np.array(recon_data['lat']),
     17          np.array(recon_data['u']), np.array(recon_data['v']),
---> 18          np.array(recon_data['peak']), transform=ccrs.PlateCarree())
     19 
     20 # Add text (aligned to the right); save the returned object so we can manipulate it.

/Users/rmay/miniconda3/envs/py35/lib/python3.5/site-packages/cartopy/mpl/geoaxes.py in barbs(self, x, y, u, v, *args, **kwargs)
   1580             if x.ndim == 1 and y.ndim == 1:
   1581                 x, y = np.meshgrid(x, y)
-> 1582             u, v = self.projection.transform_vectors(t, x, y, u, v)
   1583         return matplotlib.axes.Axes.barbs(self, x, y, u, v, *args, **kwargs)
   1584 

lib/cartopy/_crs.pyx in cartopy._crs.CRS.transform_vectors (lib/cartopy/_crs.c:6489)()

ValueError: x, y, u and v arrays must be the same shape
```

It automatically meshgrids x,y, but doesn't do anything with the vector data. In this case, I'm not even sure it's correct to mesh x,y, but I'm not sure what transform_vectors really needs.

**Comment by QuLogic** [comment]
Can you post a (short) set of data for this (even if it's just random stuff)?

**Comment by ajdawson** [comment]
This is a simple fix (already applied for `quiver` actually), see #807.

**Comment by dopplershift** [comment]
@QuLogic given #807, do you still want some sample input?

**Comment by QuLogic** [comment]
Maybe just see if that fix works for you.

**Comment by dopplershift** [comment]
Fixes it perfectly.

**Comment by ajdawson** [comment]
Fixed by #807
